### PR TITLE
check ledger keypair and add test

### PR DESCRIPTION
**Pull request by alexmat2on**
Signed-off-by: Alexander Matson <alexmat2on@protonmail.com>

### Description of the Change

Check during Irohad initialization that the keypair passed in the command line is present in the existing ledger state. Also added a test in `irohad_test` to test this function. 

### Benefits

If a keypair is passed then it is guaranteed that there is a peer that is using it. 

### Possible Drawbacks 

None

### Usage Examples or Tests *[optional]*

`irohad_test.cpp`

### CODE REVIEW

File: irohad/main/application.cpp [hunk 1]
```diff
@@ -324,6 +325,18 @@ Irohad::RunResult Irohad::restoreWsv() {
   };
 }
 
+Irohad::RunResult Irohad::validateKeypair() {
+  auto peers = storage->createPeerQuery() | [this](auto &&peer_query) {
+    return peer_query->getLedgerPeerByPublicKey(keypair.publicKey());
+  };
+  if (not peers) {
+    return expected::makeError(
+        "Failed to locate peer with provided public key!");
```
Comment (MBoldyrev) [inline irohad/main/application.cpp]: From this message it is not clear, which public key is it about. Maybe rephrase like
```suggestion
        "There is no peer in the ledger with my public key!");
```

Comment (alexmat2on) [reply irohad/main/application.cpp]: Thanks, these suggestions sound good!

**Comment by lebdron** [comment]
Removed the test because initialization does not fail on non-matching public key anymore.
